### PR TITLE
Feat(eos_cli_config_gen): Add pim ipv4 border router 

### LINKED ISSUE #3567: Feat(eos_cli_config_gen): add support for multicast border-router
### Enhancement summary

add ip igmp border-router support to interface type ethernet, port-channel and vlan

EOS doc : https://www.arista.com/en/um-eos/eos-protocol-independent-multicast#xx1151891

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

multicast use case
sample configuration:
interface Ethernet1
  pim ipv4 border-router
  

### Describe the solution you would like

eos_cli_config_gen data model

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by philippebureau**
## Change Summary

Add pim ipv4 border router to Ethernet, port-channel and vlan interfaces

## Related Issue(s)

Fixes #3567 

## Component(s) name

`eos_cli_config_gen`

## Proposed changes
new data model:
```
ethernet_interfaces:
  - name: Ethernet1
    pim:
      ipv4:
        border_router: <bool>  
```

## How to test
see molecule test

### User Checklist
- N/A

### Repository Checklist

- [ x ] My code has been rebased from devel before I start
- [ x ] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ x ] My change requires a change to the documentation and documentation have been updated accordingly.
- [ x ] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ethernet_interfaces.schema.yml [hunk 1]
```diff
@@ -460,6 +460,9 @@ keys:
             ipv4:
               type: dict
               keys:
+                border_router:
+                  type: bool
+                  description: Configure pim border router. EOS default is false
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ethernet_interfaces.schema.yml]: Add a `.` at the end. And PIM in caps (everywhere)

Comment (philippebureau) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ethernet_interfaces.schema.yml]: done

**Review by ClausHolbechArista — APPROVED** [review]
LGTM!

**Review by carl-baillargeon — APPROVED** [review]
LGTM. Tested locally.

**Pull request merged**